### ISSUE TITLE
Negative values for births and deaths

### ISSUE THREAD

**Issue by christian-oreilly** [scikit-tda/ripser.py#88]
I apologize in advance if this is explained somewhere or if it is a trivial question. I admit being very new to topological analyses. Anyway, I could not find anything (e.g., some normalization or transformation) that would explain the fact that I obtain negative values for births and deaths on the persistence plot I obtained with Ripser. This happens using the following code:

```
from ripser import Rips
rips = Rips(maxdim=4)
diagrams = rips.fit_transform(data, distance_matrix=True)
rips.plot(diagrams)
```
which produced the following figure: https://pasteboard.co/IGNZumH.png

Let me know if this is a bug, in which case I'll share the data that are used in this example so that you can reproduce the issue. If it is not a bug, maybe the documentation can highlight more clearly the fact that the plots generated are not standard persistence plots (i.e., with positive births and deaths) and in which way they depart from standard persistence plots.

**Comment by sauln** [comment]
oh very weird. Thanks for bringing this up.

Does your distance matrix have negative values?

**Comment by christian-oreilly** [comment]
Right... I was using a correlation matrix, so you are totally right... with both positives and negative values. My bad. If it is expecting a distance metric it is not surprising that it would give weird results. Maybe I could suggest testing for positivity and throwing a warning if the input matrix has negative values... for dumb-proofing the function call and helping spotting eventual issues :) Thanks for the fast feedback and for your time.

**Comment by sauln** [comment]
Phew! I'm glad that was the problem. 

The test and warning is a great idea and would probably save others in the future.  If you'd like to submit a pull request with that implementation, I would readily review and merge!

**Comment by ctralie** [comment]
Note that our lower star image filtration often applies to negative scalar
images, so that would trip the warning erroneously every time. It could be
messy to deal with that

**Comment by christian-oreilly** [comment]
@sauln I can have a look at it.

@ctralie Could you provide an example of a function call that relates to that? Not knowing that much about ripser, it could take me some time to figure out what objects/functions/methods are related to your "lower start image filtration".

**Comment by ctralie** [comment]
We have two notebooks showing this use case

https://github.com/scikit-tda/ripser.py/blob/master/docs/notebooks/Lower%20Star%20Image%20Filtrations.ipynb

https://github.com/scikit-tda/ripser.py/blob/master/docs/notebooks/Lower%20Star%20Time%20Series.ipynb

Because of this, I would vote against putting a warning for negative
values, but thank you for offering!

**Comment by christian-oreilly** [comment]
Alright. Nice notebook @ctralie 

So, no change for the moment. At least this ticket provides a minimum of documentation on the existence of negative birth and death time.
